bin/as-recon-discover-services.tengo

```diff
@@ -0,0 +1,195 @@
+#! /usr/bin/env arsenic
+
+// arsenic := import("arsenic")
+cobra := import("cobra")
+os := import("os")
+os2 := import("os2")
+filepath := import("filepath")
+check_err := import("check_err")
+nmap := import("nmap")
+script := import("script")
+log := import("log")
+
+topPorts := 0
+targets := []
+
+setupHost := func(target) {
+  baseDir := ""
+  if filepath.dir_exists("hosts") {
+    baseDir = filepath.join("hosts", target)
+  }
+
+  err := os.mkdir_all(filepath.join(baseDir, "loot"), 0755)
+  check_err(err)
+
+  err = os.mkdir_all(filepath.join(baseDir, "recon"), 0755)
+  check_err(err)
+
+  return baseDir
+}
+
+defaultScanner := func(target) {
+  return nmap.scanner().
+    stylesheet("/static/nmap.xsl").
+    open().
+    Pn().
+    n().
+    system_dns().
+    targets(target)
+}
+
+rootCmd := cobra.root_cmd("as-recon-discover-services.tengo", "Run nmap scans on targets")
+rootCmd.enable_completion()
+
+rootCmd.persistent_flags.string_slicep("targets", "t", [], "The target to scan. Can be used multiple times to specify multiple targets")
+rootCmd.persistent_flags.stringp("target-file", "f", "", "The file containing the targets to scan.")
+rootCmd.persistent_flags.int("top", 0, "Scan # most common ports")
```

Comment:
what if we could accept a percentage here as well? and added a bottom command. so we could do top 50% and bottom 50% separately.

i guess that would only work for 50%, and not other percentages. we'd need something like `port-slice`  and make that accept ranges like 1-10000, or 1-25%. this would be pretty neat for staged scans and breaking up large scopes across multiple scanning hosts.

Reply:
I can definitely see a need for that. I say we can try that but not for this PR.